apisix/discovery/consul_kv.lua

```diff
@@ -230,6 +240,70 @@ local function update_application(server_name_prefix, data)
 end
 
 
+local function read_dump_srvs()
+    local data, err = util.read_file(dump_params.path)
+    if not data then
+        log.warn("read dump file get error: ", err)
+        return
+    end
+
+    core.log.info("read dump file: ", data)
+    data = util.trim(data)
+    if #data == 0 then
+        core.log.error("dump file is empty")
+        return
+    end
+
+    local entity, err  = core.json.decode(data)
+    if err then
+        core.log.error("decoded dump data got error: ", err, ", file content: ", data)
+        return
+    end
+
+    if not entity.services or not entity.last_update then
+        core.log.error("decoded dump data miss fields, file content: ", data)
+        return
+    end
+
+    if dump_params.expire ~= 0  and (entity.last_update + dump_params.expire) < ngx.time() then
+       core.log.warn("dump file: ", dump_params.path,
+         " had expired, ignored it")
+        return
+    end
+
+    applications = entity.services
+    core.log.info("load dump file into memory success")
+end
+
+
+local function write_dump_srvs()
+    local entity = {
+        services = applications,
+        last_update = ngx.time(),
+        expire = dump_params.expire, -- later need handle it
+    }
+    local data = core.json.encode(entity)
+    local succ, err =  util.write_file(dump_params.path, data)
+    if not succ then
+        core.log.error("write dump into file got error: ", err)
+    end
+end
+
+
+local function show_dump_file()
+    if not dump_params then
+        return 200, "dump params is nil"
```

Comment:
Should we use a non-200 status code when an error happened?
And I think we need to test for error in this function.

Reply:
1. You are right, we should return 500 status code when an error happened
2. Add Test for `show_dump_file` function when get an error ?

Reply:
> Add Test for show_dump_file function when get an error ?

Yes